Python/fileutils.c

```diff
@@ -2450,6 +2473,41 @@ _Py_set_blocking(int fd, int blocking)
     return -1;
 }
 #else   /* MS_WINDOWS */
+int
+_Py_get_blocking(int fd)
+{
+    HANDLE handle;
+    DWORD mode;
+
+    handle = _Py_get_osfhandle(fd);
+    if (handle == INVALID_HANDLE_VALUE)
+        return -1;
+
+    if (!GetNamedPipeHandleState(handle, &mode, NULL, NULL, NULL, NULL, 0)) {
```

Comment:
We link directly to [W]ide-character functions instead of relying on the macros. It's good to be consistent about this even though it doesn't technically matter here.

```suggestion
    if (!GetNamedPipeHandleStateW(handle, &mode, NULL, NULL, NULL, NULL, 0)) {
```